articles/prettier-setup/index.md

```diff
@@ -0,0 +1,188 @@
+
+```
+  echo {}> .prettierrc.json         ... File A
+```
+
+```
+  # Ignore artifacts:          
+  package-lock.json
+  .next
+  node_modules/                     ... File B
+```
+
+- Once you're done with writing your code files in the project, it's now time to format all the files you have.
+
+```
+  npx prettier --write .    (using NPM)
+  yarn prettier --write .   (using Yarn)
+```
+![format image](./format.png)
+
+### How to configure Prettier to work with ESLint?
+
+- ESLint is a programming tool that analyzes and checks source code. Apart from removing programming, indentation and formatting errors, it also removes bugs and undesirable practices in code formatting. The linting feature of ESLint & efficient formatting of Prettier is a deadly combination that increases productivity manifolds. To do the same, first install ESLint on your system. Here is a wonderful [tutorial](https://www.section.io/engineering-education/node-eslint/#eslint-and-node) you can refer to, for installing ESLint.
+
+- After installing Prettier and ESLint, install the following packages:
+
+```
+  npm i prettier eslint-config-prettier eslint-plugin-prettier -D
+```
+![plugin image](./eslintconf.png)
+
+- Now, Update “extends” in your .eslintrc file as follows:
+
+```
+  "extends": [ "airbnb", "plugin:prettier/recommended" ]
+```
+- If you want to modify the default Prettier configuration, just create a **.prettierrc** file in *“eslint-app”* directory, for example:
+
+```
+  {
+    // Use single quotes instead of double quotes.
+    "singleQuote": false,
+
+    /**
+     * Print trailing commas wherever possible.
+     * Valid options:
+     *   - "none" - no trailing commas
+     *   - "es5" - It prints trailing commas where valid in ES5 (ex: objects, arrays, etc)
+     *   - "all" - It prints trailing commas wherever possible (ex: function arguments)
+     */
+    "trailingComma": "es5",
+
+    /**
+     * To not print the corresponding semicolons, except at the beginning of lines which may need them.
+     * Valid options:
+     * - true - It adds a semicolon at the end of every line
+     * - false - It only adds semicolons at the very beginning of lines that introduce ASI failures
+     */
+    "noSemi": true,
+
+    /**
+     * To not print the spaces between brackets.
+     * If false, puts the > of a multi-line JSX element at the end of the next line instead of being
+     * alone on the last
+     */
+    "jsxBracketSameLine": false,
+
+     // add many more as required!
+
+  }
+```
+![config. image](./prettierconf.png)
+
+### How to format all your code with just one-click?
+- Instead of running Prettier from the command line (prettier --write), checking formatting in CI, or running it from the editor, you can run Prettier as a pre-commit hook as well. This makes sure all your commits are formatted, without having to wait for your CI build to finish in a single click!
```

Comment:
This sentence seems like a run on sentence - perhaps we can structure it a bit differently.